### ISSUE TITLE
Qustion about A3C "apply_gradients" function

### ISSUE THREAD

**Issue by mashoujiang** [hongzimao/pensieve#9]
Hi,

I am reading your a3c.py and multi_agent.py code, the function "apply_gradients(self, actor_gradients)" really makes me confuse. 
    def apply_gradients(self, actor_gradients):
        return self.sess.run(self.optimize, feed_dict={
            i: d for i, d in zip(self.actor_gradients, actor_gradients)
        })
which will inference: 
# Optimization Op
        self.optimize = tf.train.RMSPropOptimizer(self.lr_rate).\
            apply_gradients(zip(self.actor_gradients, self.network_params))

This self.actor_gradients op is not even a placeholder, why can you just feed_dict it with the fixed value? which is got from get_gradients().

Thanks!

**Comment by hongzimao** [comment]
I admit it's not the most elegant way of implementing `apply_gradient` and sorry for the confusion. But you indeed can pass the gradient this way. Feel free to poke around the code and verify. Thanks!

**Comment by mashoujiang** [comment]
@hongzimao . maybe that is not a significant problem. :) . Thanks for your reply anyway.
